samples/Basic_Delegates/app/application.cpp

```diff
@@ -9,19 +14,86 @@ public :
 		pinMode(ledPin, OUTPUT);
 	};
 	bool setTimer(int reqInterval) {
-		if (reqInterval <= 0) return false;
+		if (reqInterval <= 0) {
+			return false;
+		}
 		ledInterval = reqInterval;
 		return true;
 	}
-	void blink(bool reqRun) {
+
+	// This example show the way delegates have been used in Sming in the past.
+	void blinkOldDelegate(bool reqRun) {
+		if (reqRun) {
+			ledTimer.initializeMs(ledInterval, TimerDelegate(&LedBlinker::ledBlink, this)).start();
+		}
+		else {
+			ledTimer.stop();
+		}
+	}
+
+	// This example shows how to use a plain old ordinary function as a callback
+	void callPlainOldOrdinaryFunction(bool reqRun) {
+		if (reqRun) {
+			ledTimer.initializeMs(ledInterval, TimerDelegateStdFunction(plainOldOrdinaryFunction)).start();
+			// or just
+			// ledTimer.initializeMs(ledInterval, plainOldOrdinaryFunction).start();
+		}
+		else {
+			ledTimer.stop();
+		}
+	}
+
+
+	// Sming now allows the use of std::function
+	// This example shows how to use a lamda expression as a callback
+	void callLamda(bool reqRun) {
+		if (reqRun) {
+			int foo = 123;
+			ledTimer.initializeMs(ledInterval, 
+				[&]	// capture everything by reference - so foo will be available inside the lamda
+				()  // No parameters to the callback
+				{
+					debugf("lamda Callback ");
+					if (foo > 123) {
```

Comment:
@frankdownunder In the if condition shouldn't there be an equal sign instead of bigger ?

Reply:
Ive fixed that now.